Sources/TuistKit/Cache/CacheController.swift

```diff
@@ -188,4 +198,33 @@ final class CacheController: CacheControlling {
 
         _ = try cache.store(hash: hash, paths: FileHandler.shared.glob(outputDirectory, glob: "*")).toBlocking().last()
     }
+
+    fileprivate func buildAndCacheBundle(path: AbsolutePath,
+                                         target: GraphTarget,
+                                         configuration: String,
+                                         hash: String) throws
+    {
+        let outputDirectory = try FileHandler.shared.temporaryDirectory()
+        defer {
+            try? FileHandler.shared.delete(outputDirectory)
+        }
+
+        if path.extension == "xcworkspace" {
+            try bundleArtifactBuilder.build(
+                workspacePath: path,
+                target: target.target,
+                configuration: configuration,
+                into: outputDirectory
+            )
+        } else {
+            try bundleArtifactBuilder.build(
+                projectPath: path,
+                target: target.target,
+                configuration: configuration,
+                into: outputDirectory
+            )
+        }
+
+        _ = try cache.store(hash: hash, paths: FileHandler.shared.glob(outputDirectory, glob: "*")).toBlocking().last()
```

Comment:
What about adjusting the `build` method in `bundleArtifactBuilder` to return the path to the bundle inside the output directory? Otherwise you have the globbing logic in two places, here, and inside the bundle instance.

Reply:
Makes sense but `build` method is declared on `CacheArtifactBuilding` protocol. Changing its signature will also affect `CacheFrameworkBuilder` and `CacheXCFrameworkBuilder`